alacritty/src/window_context.rs

```diff
@@ -183,9 +188,12 @@ impl WindowContext {
             self.display.pending_update.dirty = true;
         }
 
-        // Live title reload.
-        if !config.window.dynamic_title || old_config.window.title != config.window.title {
-            self.display.window.set_title(&config.window.title);
+        // Live title reload only when title wasn't provided on command line.
+        //
+        // We know that the title was provided on the command line when dynamic_title is disaabled
+        // and old_config title isn't equal to the current one.
+        if !config.window.dynamic_title && self.identity.title != old_config.window.identity.title {
```

Comment:
So when I have a title specified on the CLI and have the same in the config, it will overwrite it when I change the config?

Reply:
Yeah, that is the case when it will be overwritten anyway. If we want to handle it as well we should pass `Option<Idenetity>` and overwrite config right in the `WindowContext`.

Reply:
Your approach would be to just not bother because not worth? I'd be fine with it tbh.

Reply:
Yeah, pretty much it. I just override title when it isn't the one you passed on CLI which shouldn't be equal to default title.